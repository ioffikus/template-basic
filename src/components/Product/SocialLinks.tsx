import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import cx from 'classnames';
import Svg from 'src/components/Svg';

interface IOwnProps {}

interface IConnectedState {
  ui: Alicanto.Models.UI;
}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState, ownProps: IOwnProps) => ({
  ui: state.ui,
});

const WrappedSocial = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Social extends React.Component<IProps, IState> {
    render() {
      const itemsUI = this.props.ui.routes.productCard.socialLinks.items;
      const visibleItems = _.transform(
        itemsUI,
        (result, value, key) => {
          const isVisible = value.settings.isVisible;
          if (isVisible) {
            result.push({ ...value, name: key });
          }
        },
        [],
      );
      return (
        <div className="root">
          <style jsx>{`
            .list {
              display: flex;
              align-items: flex-end;
              justify-content: flex-end;
              margin: 0;
              padding: 0;
              list-style-type: none;
            }
            .root :global(.link) {
              font-size: 0;
            }
            .item {
              display: flex;
              align-items: center;
              justify-content: center;
              width: 25px;
              height: 25px;
            }
            .root :global(.icon-facebook) {
              fill: #008fff;
              width: 12px;
              height: 17px;
            }
            .root :global(.icon-twitter) {
              fill: #009cff;
              width: 11px;
              height: 17px;
            }
            .root :global(.icon-vkontakte) {
              fill: #4c75a3;
              width: 15px;
              height: 12px;
            }
            .root :global(.icon-instagram) {
              fill: #ff0070;
              width: 14px;
              height: 14px;
            }
          `}</style>
          <ul className="list">
            {visibleItems.map(item => {
              const link = item.i18n.defaultText;
              const name = item.name;
              return (
                <li className="item" key={name}>
                  <a href={link} target="_blank" rel="noopener noreferrer" className="link">
                    <Svg
                      name={name}
                      className={cx('icon', {
                        [`icon-${name}`]: name,
                      })}
                    />
                    {name}
                  </a>
                </li>
              );
            })}
          </ul>
        </div>
      );
    }
  },
);

export default WrappedSocial;
