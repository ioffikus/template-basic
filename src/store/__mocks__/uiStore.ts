import { IUIState } from 'src/ducks/ui';
import fs from 'fs';
import yaml from 'js-yaml';

const globalConfig = yaml.safeLoad(fs.readFileSync('src/configs/global.yml', 'utf8'));
export const ui = globalConfig.ui;

const state: IUIState = {
  ...ui,
  isRequestError: false,
  isRequesting: false,
};

export default state;
