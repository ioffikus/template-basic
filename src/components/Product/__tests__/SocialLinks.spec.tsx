import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import uiStore from 'src/store/__mocks__/uiStore';
import SocialLinks from '../SocialLinks';

const mockStore = configureMockStore([thunk]);

describe(null, () => {
  it('render() correctly', () => {
    const store = mockStore({
      ui: uiStore,
    });
    const wrapper = shallow(<SocialLinks />, { context: { store } }).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
