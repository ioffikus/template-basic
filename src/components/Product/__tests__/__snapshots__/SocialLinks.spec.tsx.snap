// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correctly 1`] = `
<div
  className="jsx-3266447429 root"
>
  <JSXStyle
    css=".list.jsx-3266447429{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:flex-end;-webkit-box-align:flex-end;-ms-flex-align:flex-end;align-items:flex-end;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;margin:0;padding:0;list-style-type:none;}.root.jsx-3266447429 .link{font-size:0;}.item.jsx-3266447429{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:25px;height:25px;}.root.jsx-3266447429 .icon-facebook{fill:#008fff;width:12px;height:17px;}.root.jsx-3266447429 .icon-twitter{fill:#009cff;width:11px;height:17px;}.root.jsx-3266447429 .icon-vkontakte{fill:#4c75a3;width:15px;height:12px;}.root.jsx-3266447429 .icon-instagram{fill:#ff0070;width:14px;height:14px;}"
    styleId="3266447429"
  />
  <ul
    className="jsx-3266447429 list"
  >
    <li
      className="jsx-3266447429 item"
      key="facebook"
    >
      <a
        className="jsx-3266447429 link"
        href="https://www.facebook.com/internationalrelationsrtu"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Svg
          className="icon icon-facebook"
          name="facebook"
        />
        facebook
      </a>
    </li>
    <li
      className="jsx-3266447429 item"
      key="twitter"
    >
      <a
        className="jsx-3266447429 link"
        href="https://twitter.com/RTU_LV"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Svg
          className="icon icon-twitter"
          name="twitter"
        />
        twitter
      </a>
    </li>
    <li
      className="jsx-3266447429 item"
      key="vkontakte"
    >
      <a
        className="jsx-3266447429 link"
        href="http://vk.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Svg
          className="icon icon-vkontakte"
          name="vkontakte"
        />
        vkontakte
      </a>
    </li>
    <li
      className="jsx-3266447429 item"
      key="instagram"
    >
      <a
        className="jsx-3266447429 link"
        href="http://instagram.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Svg
          className="icon icon-instagram"
          name="instagram"
        />
        instagram
      </a>
    </li>
  </ul>
</div>
`;
